app/models.py

```diff
@@ -471,3 +486,8 @@ def filter_null_value_fields(obj):
     return dict(
         filter(lambda x: x[1] is not None, obj.items())
     )
+
+
+def generate_new_service_id():
+    # TODO: Decide what we want G7 service IDs to look like and implement
```

Comment:
I think the publish flow is still very much a work in progress with a fair amount of iteration to be done still (e.g. I think the publish endpoint will need to be aware of framework status and do different things with a draft depending on the status of it's framework).  I think the benefits of having end-to-end functionality outweigh the problems with having a temporary patch in here, so I'd argue to leave the method in.

The point about validation is a good one though - UUID won't match the current rules, and the fact that the tests pass show that we're missing at least one test on this - I'll fix it up.